Azure Resources Page (#46)
UWB-ACM/Hackathon-Docs-2020: _docs/azure_resources.md

permalink: /azure_resources.html
---

# Azure Resources, SDK Links, Tutorials, Helpful Tidbits, and Permissions Management

## SDKs and Tools
### SDKs
Azure offers two types of SDKs. Unified SDKs are built on a "common core" and are recommended for all new projects. Standard SDKs are avilable for older projects or languages unsupported by Unified SDKs.

> - [Unified SDKs](https://azure.github.io/azure-sdk/releases/latest/index.html#java-packages) available for: .NET, Java, Python, TypeScript/JavaScript

Reviewer: ```suggestion
- [Unified SDKs](https://azure.github.io/azure-sdk/releases/latest/index.html) available for: .NET, Java, Python, TypeScript/JavaScript
```
might as well link using the top of the page

Author: Good catch, thanks

- [Standard .NET SDKs](https://azure.github.io/azure-sdk/releases/latest/all/dotnet.html) (also included with Visual Studio 2019)
- [Standard Java SDKs](https://azure.github.io/azure-sdk/releases/latest/all/java.html) 
- [Standard Python SDKs](https://azure.github.io/azure-sdk/releases/latest/all/python.html) 
- [Standard TypeScript/JavaScript SDKs](https://azure.github.io/azure-sdk/releases/latest/all/js.html) 
- [Standard PHP SDKs](https://packagist.org/packages/microsoft/) 
- [Standard Go SDKs](https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-install) 
- [Azure SDKs Page](https://azure.microsoft.com/en-us/downloads/)

### Tools
- [Visual Studio 2019](https://visualstudio.microsoft.com/) has excellent integrations with Azure services, especially when using C#/.NET.

> - [Azure Visual Studio](https://azure.microsoft.com/en-us/products/visual-studio/)

Reviewer: seems like a redundant link?

Author: Yeah, definitely

- [Azure Cloud Shell](https://docs.microsoft.com/en-us/azure/cloud-shell/overview)
- [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/)
- [Azure Data Studio](https://docs.microsoft.com/en-us/sql/azure-data-studio/download-azure-data-studio?view=sql-server-2017)
- [Azure Tools Page](https://azure.microsoft.com/en-us/product-categories/developer-tools/)

## Tutorials
- [Getting Started with Azure](https://azure.microsoft.com/en-us/get-started/)
- [Microsoft Learn - Azure](https://docs.microsoft.com/en-us/learn/azure/)
- [Azure Documentation](https://docs.microsoft.com/en-us/azure/?product=featured)
- [Azure 15-Minute Tutorials](https://tutorials.visualstudio.com/)
- [Getting Started with Visual Studio](https://visualstudio.microsoft.com/vs/getting-started/)
- [Developing Azure Functions with Visual Studio](https://docs.microsoft.com/en-us/azure/azure-functions/functions-develop-vs)
- [Deploying Azure Applications with VSCode](https://code.visualstudio.com/docs/azure/deployment)
- [The Developer's Guide to Azure](https://azure.microsoft.com/en-us/campaigns/developer-guide/)

## Helpful Tidbits

> - Most resources on Azure require supporting resources to be provisioned alongside the main offering. For example, provisioning a VM will also require a Storage Account, Virtual Networking, etc. These resources will remain provisioned *even if you delete the main resource.* Remember to delete **all** resources when you are done using them to avoid charges.

Reviewer: To add to this, this is why "resource groups" are a thing. Under a subscription you can create many resource groups that represent individual projects. When it comes time to delete resources, you can just delete the whole resource group.

- The protocols that Azure supports for communicating with its services differs from that of AWS or GCP. Refer to the documentation for information specific to each service.

## Security and Permissions Management
### Role-Based Access Control (RBAC)
Access to Azure resources is managed through RBAC. RBAC allows you to define how to grant access. It can be done based on resource type, subscriptions, resource groups, or the specific resources themselves. Before continuing, see the [Azure documentation page on RBAC](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview). RBAC uses the concept of a **role assignment** to assign permissions. Role assignments have three parts.

> - **Security Principals** are objects that represent the entity that must be granted permissions. There are four tupes of supportal prinipals: users, groups, service prinicpals, and managed identities. See the documentation page for more info on what each type of security principal is for.

Reviewer: ```suggestion
- **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the documentation page for more info on what each type of security principal is for.
```

Remember to delete the **entire** resource group when deleting resources in order to avoid getting charged for vestigial resources.

### Take Note of Supported Protocols
The protocols that Azure supports for communicating with its services differs from that of AWS or GCP. Depending on the application you wish to build, the supported protocols may affect your architecture. Refer to the documentation for information specific to each service.

## Security and Permissions Management
### Role-Based Access Control (RBAC)
Access to Azure resources is managed through RBAC. RBAC allows you to define how to grant access. It can be done based on resource type, subscriptions, resource groups, or the specific resources themselves. Before continuing, see the [Azure documentation page on RBAC](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview). RBAC uses the concept of a **role assignment** to assign permissions. Role assignments have three parts.

> - **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the documentation page for more info on what each type of security principal is for.

Reviewer 2: Add embedded link to docs page for good measure
```suggestion
- **Security Principals** are objects that represent the entity that must be granted permissions. There are four types of security principals: users, groups, service principals, and managed identities. See the [documentation page](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview#security-principal) for more info on what each type of security principal is for.
```
